### ISSUE TITLE
Duplicate rbac roles

### ISSUE THREAD

**Issue by s12chung** [kubernetes-sigs/kubebuilder#558]
when placing two different comments for the same resource, I get duplicate/overlaying rbac roles.

ex:
```go
// +kubebuilder:rbac:groups="",resources=events,verbs=get
// +kubebuilder:rbac:groups="",resources=events,verbs=get,list
```

I get:
```yaml
- apiGroups:
   - ""
   resources:
   - events
   verbs:
   - get
- apiGroups:
   - ""
   resources:
   - events
   verbs:
   - get
   - list
```

or if i duplicate the comment, I get two entries. I expect the entries to merge together

**Comment by fanzhangio** [comment]
I might miss something, what is the particular reason why not merge the two RABC comments into one for the same resource in your case.

**Comment by s12chung** [comment]
we have two controllers that watch the same resource, so i think it'd make sense to have a comment for each controller.

**Comment by droot** [comment]
Expectation is correct, It will be nice for the tool to de-dupe the RBAC roles. I think our current tooling just has a flat list, converting it to a map key'd by GVK can help with this. 

For now, as a workaround, I would recommend having merged comment.

**Comment by DirectXMan12** [comment]
Is this actually breaking anything?  If not, there's not really much of a reason to care.

**Comment by cassianoleal** [comment]
It can be confusing when troubleshooting, makes it look like the tool is not working properly and might cause wasted time and effort.

**Comment by DirectXMan12** [comment]
ack.  closing this -- we've got an issue open in controller-tools to fix (kubernetes-sigs/controller-tools#235)
